kythe/rust/extractor/src/bin/save_analysis.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 The Kythe Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::Result;
+use std::path::PathBuf;
+
+/// Generates a save analysis in `output_dir`
+///
+/// * `arguments` - The Bazel arguments extracted from the extra action protobuf
+/// * `output_dir` - The base directory to output the save_analysis
+pub fn generate_save_analysis(arguments: Vec<String>, output_dir: PathBuf) -> Result<()> {
+    let rustc_arguments = generate_arguments(arguments, &output_dir)?;
+    let _input_files = kythe_rust_extractor::generate_analysis(rustc_arguments, output_dir)
```

Comment:
I must have missed this earlier, but the extractor shouldn't be doing any analsysis other than determining what source files are required.

Reply:
Unfortunately, we need to do some extra work during extraction because intermediate build results are tied to the exact compiler revision used to build them. This is like the problem we have with modules in Clang/PCHs, but worse because there's no fallback for compatibility.

Reply:
I agree this is the expedient solution (and that Wyatt should proceed), but it's neither the only solution nor a particularly robust solution as the generated JSON is *also* tightly versioned (and potentially insufficient).

Reply:
The difference with the generated JSON is that we control the versioning with the version of `rustc_save_analysis` that we use in the extractor which is what controls the `save_analysis` version, compared to the `.rlib` versioning being controlled by a team's choice of their Rust compiler version. The `rls_data` crate, which houses the `Analysis` data structure, has also been stable since April 2019.

But I also agree that I should be thinking about a way to transition off of `save_analysis` for the future, especially if the feature will eventually be deprecated.